common/lib/xmodule/xmodule/course_metadata_utils.py

```diff
@@ -222,3 +223,47 @@ def may_certify_for_course(certificates_display_behavior, certificates_show_befo
         or certificates_show_before_end
     )
     return show_early or has_ended
+
+
+def sorting_score(start, advertised_start, announcement):
+    """
+    Returns a tuple that can be used to sort the courses according
+    the how "new" they are. The "newness" score is computed using a
+    heuristic that takes into account the announcement and
+    (advertised) start dates of the course if available.
+
+    The lower the number the "newer" the course.
+    """
+    # Make courses that have an announcement date shave a lower
+    # score than courses than don't, older courses should have a
+    # higher score.
+    announcement, start, now = sorting_dates(start, advertised_start, announcement)
+    scale = 300.0  # about a year
+    if announcement:
+        days = (now - announcement).days
+        score = -exp(-days / scale)
+    else:
+        days = (now - start).days
+        score = exp(days / scale)
+    return score
+
+
+def sorting_dates(start, advertised_start, announcement):
+    """
+    Utility function to get datetime objects for dates used to
+    compute the is_new flag and the sorting_score.
+    """
+    announcement = announcement
+    if announcement is not None:
+        announcement = announcement
+
+    try:
+        start = dateutil.parser.parse(advertised_start)
+        if start.tzinfo is None:
+            start = start.replace(tzinfo=UTC())
+    except (ValueError, AttributeError):
+        start = start
```

Comment:
Is this just the equivalent of "pass", but someone was not so clearly trying to say that start is just the original param?
Spoke to @nasthagiri and it depends on what line the exception is happening.  I think Nimisha may look at this with a separate PR.